File: 20.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """

        stack = []
        brackets_map = { ')': '(', '}' : '{', ']' : '['}

        for char in s:
            if char in brackets_map:
                if (len(stack) == 0 or stack[-1] != brackets_map[char]):
                    return False
                stack.pop()
            else:
                stack.append(char)
        
        return len(stack) == 0

File: 20.ts
function isValid(s: string): boolean {
    const stack: string[] = [];
    const brackets = { ')': '(', '}': '{', ']': '[' }

    for (const char of s) {
        if (brackets[char]) {
            if (!stack || stack[stack.length - 1] !== brackets[char]) {
                return false
            }
            stack.pop()
        } else {
            stack.push(char)
        }
    }

    return stack.length === 0;
};